diff --git a/cliche/install_generator.py b/cliche/install_generator.py
index b9b3b81..02a50fc 100644
--- a/cliche/install_generator.py
+++ b/cliche/install_generator.py
@@ -7,6 +7,7 @@ import re
 import time
 import json
 import glob
+from pathlib import Path
 
 sys.cliche_loaded_modules__ = set(sys.modules)
 sys.cliche_ts__ = time.time()
@@ -61,7 +62,10 @@ if use_timing:
 
 if any_change:
     cache = new_cache
-    with open("{{bin_name}}.json", "w") as f:
+    cache_dir = os.path.dirname(str(Path.home()) + "{{bin_name}}.json")
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir)
+    with open(str(Path.home()) + "{{bin_name}}.json", "w") as f:
         json.dump(cache, f)
 
 function_to_imports = {}
